src/mono/wasm/Wasm.Build.Tests/IcuTests.cs

```diff
@@ -232,15 +232,19 @@ public void FullIcuFromRuntimePackWithInvariant(BuildArgs buildArgs, bool invari
     }
 
     [Theory]
-    [MemberData(nameof(FullIcuWithICustomIcuTestData), parameters: new object[] { false, RunHost.NodeJS | RunHost.Chrome })]
+    [MemberData(nameof(FullIcuWithICustomIcuTestData), parameters: new object[] { false, RunHost.NodeJS | RunHost.Chrome })] 
     [MemberData(nameof(FullIcuWithICustomIcuTestData), parameters: new object[] { true, RunHost.NodeJS | RunHost.Chrome })]
     public void FullIcuFromRuntimePackWithCustomIcu(BuildArgs buildArgs, bool fullIcu, RunHost host, string id)
     {
         string projectName = $"fullIcuCustom_{fullIcu}_{buildArgs.Config}_{buildArgs.AOT}";
         bool dotnetWasmFromRuntimePack = !(buildArgs.AOT || buildArgs.Config == "Release");
 
         buildArgs = buildArgs with { ProjectName = projectName };
-        buildArgs = ExpandBuildArgs(buildArgs, extraProperties: $"<WasmIcuDataFileName>{s_customIcuPath}</WasmIcuDataFileName><WasmIncludeFullIcuData>{fullIcu}</WasmIncludeFullIcuData>");
+        string customFile = fullIcu ? "icudt_nonexisting.dat" : s_customIcuPath;
```

Comment:
Selecting the filename based on `fullIcu`, and then later in the test looking for the warning based on `fullIcu` is too embedded in the details. Instead, add a new parameter to the test like `useCustomFileFromRuntimePack`.

Reply:
It's not from runtime pack -> `useCustomFile`.